bindings/rust/aws-lc-sys-template/build.rs

```diff
@@ -0,0 +1,271 @@
+            "{}{}",
+            if let Some(pfix) = prefix.to_owned() {
+                pfix
+            } else {
+                ""
+            },
+            match self {
+                OutputLib::Crypto => "crypto",
+                OutputLib::Ssl => "ssl",
+            }
+        )
+    }
+
+    fn locate_dir(&self, path: &Path) -> PathBuf {
+        path.join(Path::new(&format!("build/{}", self.libname(None))))
+            .join(get_platform_output_path())
+    }
+}
+
+fn get_platform_output_path() -> PathBuf {
+    PathBuf::new()
+}
+
+fn get_cmake_config() -> cmake::Config {
+    let pwd = env::current_dir().unwrap();
+
+    cmake::Config::new(pwd.join(AWS_LC_PATH))
+}
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+fn prefix_string() -> String {
+    format!("aws_lc_{}", VERSION.to_string().replace('.', "_"))
+}
+
+fn test_cmake_command(executable: &str) -> bool {
+    if let Ok(output) = Command::new(executable).arg("--version").output() {
+        return output.status.success();
+    }
+    false
+}
+
+fn prepare_cmake_build(build_prefix: Option<&str>) -> cmake::Config {
+    if test_cmake_command("cmake3") {
+        env::set_var("CMAKE", "cmake3");
+    } else {
+        env::set_var("CMAKE", "cmake");
+    }
+
+    let mut cmake_cfg = get_cmake_config();
+
+    if let Some(symbol_prefix) = build_prefix {
+        cmake_cfg.define("BORINGSSL_PREFIX", symbol_prefix);
+        let pwd = env::current_dir().unwrap();
+        let include_path = pwd.join(AWS_LC_PATH).join("include");
+        cmake_cfg.define(
+            "BORINGSSL_PREFIX_HEADERS",
+            include_path.display().to_string(),
+        );
+    }
+
+    cmake_cfg
+}
+
+fn build_aws_lc() -> PathBuf {
+    let mut cmake_cfg = prepare_cmake_build(Some(&prefix_string()));
+
+    cmake_cfg.build_target("crypto").build()
+}
+
+fn main() -> Result<(), String> {
+    use crate::OutputLib::Crypto;
+    use crate::OutputLibType::Static;
+
+    let manifest_dir = env::current_dir().unwrap();
+    let manifest_dir = dunce::canonicalize(&Path::new(&manifest_dir)).unwrap();
+    let prefix = prefix_string();
+
+    let bindings_file = manifest_dir.join("src").join("bindings.rs");
+
+    let builder = prepare_bindings_builder(&manifest_dir, Some(&prefix));
+    let bindings = builder.generate().expect("Unable to generate bindings.");
+    bindings
+        .write_to_file(&bindings_file)
+        .expect("Unable to write bindings to file.");
+    modify_bindings(&bindings_file, &prefix).map_err(|err| err.to_string())?;
+
+    let aws_lc_dir = build_aws_lc();
+
+    let libcrypto_dir = Crypto.locate_dir(&aws_lc_dir);
+    println!("cargo:rustc-link-search=native={}", libcrypto_dir.display());
+    println!(
+        "cargo:rustc-link-lib={}={}",
+        Static.rust_lib_type(),
+        Crypto.libname(None)
+    );
+
+    println!(
+        "cargo:include={}",
+        get_include_path(&manifest_dir).display()
+    );
+    println!("cargo:rerun-if-changed=build.rs");
```

Comment:
nit: this already works out of the box so it isn't really needed

Reply:
The default behavior seems slightly more likely to [trigger an unnecessary build](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed). However, in practice the behavior our dependencies see should be the same whether this line is present or not.

> If the build script inherently does not need to re-run under any circumstance, then emitting cargo:rerun-if-changed=build.rs is a simple way to prevent it from being re-run (otherwise, the default if no rerun-if instructions are emitted is to scan the entire package directory for changes).